### DISCUSSION TITLE
questions about IVF_PQ and sizing tool

### DISCUSSION THREAD

**Discussion by broken-blade** [milvus-io/milvus#19016] [Q&A and General discussion] [answered]
Hi,all!
I'm testing the recall rate of IVF_PQ index (milvus v1.1.1, CPU edition), and at first I use such index param to build PQ index `index_param = {
    'nlist': 4096,
    'm': 32
}` , then I found another parameter `nbits` in [this page](https://milvus.io/docs/v1.1.1/index.md#CPU), and it's default value is 8.
Here is my questions:

1. Does this `nbits` parameter represents that how many bits is needed for the value of 'nlist'?like 4096 need 12 bits.
2. Does `nbits` affects the IVF_PQ index size? if it affects, the sizing tool page seems not contains this parameter
3. I have already insert 80M vectors into milvus, and how can I know the accuracy size of memory space it needs,not a recommend value from sizing tool

**Comment by yhmo** [comment, marked as answer]
1. IVF_PQ index come from FAISS, it uses IVF method to split vectors into clusters(each cluster has a centroid), and use PQ algorithm to convert vector to binary bits(we call quantizer). The 'nlist' is parameter of IVF, the 'm' and 'nbit' are parameters of PQ. The 'm' splits each vector into m blocks. If 'nbit = 8', the index engine uses IVF to iterate 256 centroids for each block column. So, the 'nbit' is totally different from 'nlist'.
 
2. The size of IVF_PQ is :  (nb * m * nbits / 8 + 2^nbits * d * 4 + nlist * d * 4). I guess the size tool use the default value 8 to calculate.

3. You can use command `top` to list the memory usage.
